### ISSUE TITLE
misopy not finding paired mates from Tophat Bam

### ISSUE THREAD

**Issue by alig99** [yarden/MISO#31]
Hi,

I'm trying to compute the insert length distribution given a constitutive exons file.
The bam file comes from TopHat, but misopy is complaining that there are no read pairs, though Flagstat claims otherwise.  I tried --no-bam-filter but it made no difference.
I have checked the fastq files & the pairs have the same IDs except of course 1 for R1 & 2 for R2.

Any help would be greatly appreciated.  This is my command & error output.  Thanks ali

python misopy/pe_utils.py --no-bam-filter --compute-insert-len tophat2.mm10.bam misopy-0.4.6/exons/Mus_musculus.NCBIM37.65.min_1000.const_exons.gff --output-dir /misopy-0.4.6/insert-dist/
Computing insert length distribution of 1 files:
 tophat2.mm10.bam
- Using const. exons from: misopy-0.4.6/exons/Mus_musculus.NCBIM37.65.min_1000.const_exons.gff
- Outputting to: misopy-0.4.6/insert-dist
- Minimum exon size used: 500
  Getting constitutive exons...
- Input GFF: misopy-0.4.6/exons/Mus_musculus.NCBIM37.65.min_1000.const_exons.gff
- Output dir: misopy-0.4.6/insert-dist
- Output format: gff
- Including only exons greater than or equal to 500-bp
  Constitutive exons GFF was given, so not outputting another one.
  Turning off filtering of BAM reads
  Fetching reads in constitutive exons
  Mapping BAM to GFF...
- BAM: tophat2.mm10.bam
- GFF: misopy-0.4.6/exons/Mus_musculus.NCBIM37.65.min_1000.const_exons.gff
- Output file: misopy-0.4.6/insert-dist/bam2gff_Mus_musculus.NCBIM37.65.min_1000.const_exons.gff/tophat2.mm10.bam
  WARNING: misopy-0.4.6/insert-dist/bam2gff_Mus_musculus.NCBIM37.65.min_1000.const_exons.gff/tophat2.mm10.bam exists. Skipping..
  Filtered out 0 read pairs that were on same strand.
  Filtered out 0 reads that had no paired mate.
- Total read pairs: 0
  WARNING: no paired mates in tophat2.mm10.bam. Skipping...
  Are you sure the read IDs match? If your BAM paired flags are unset, try using --no-bam-filter

**Comment by yarden** [comment]
Hi,

Could you please verify that you don't have a chromosome headers mismatch (as described here http://genes.mit.edu/burgelab/miso/docs/#human-mouse-gene-models-for-isoform-centric-analyses)? 

The way to do it is to make sure that the headers in your BAM file match the headers in your input GFF. The GFF file you're using here looks like it's from Ensembl (`Mus_musculus.NCBIM37.65.min_1000.const_exons.gff`) -- does your BAM file have numeric chromosomes, e.g. `1` and not UCSC style `chr1`? Try:

`samtools view tophat2.mm10.bam | head -n 100`

What does that output? Alternatively you can do:

`samtools view -H tophat2.mm10.bam`

It should say `SN:1` for example and not `SN:chr1`

**Comment by alig99** [comment]
Hi Yarden,

Thanks so much for your quick response.

It does appear that I do have a chromosome headers mismatch.

I was aware that this could happen, but thought the Ensembl gff file had come from UCSC & therefore would have the same chr headers as the UCSC RefSeq the Bam was aligned to.  Apologies for wasting your time.  This is my first attempt at using Miso.

I don't want to realign the reads with a different RefSeq as these were aligned for me by someone else.

So have followed your instructions on how to convert to gff3 from UCSC GenePred format.

Thank you for your help

Alison

**Comment by yarden** [comment]
It's true that if you get UCSC from Ensembl it follows UCSC headers as you say, so it is confusing. The file we posted is from Ensembl directly so it follows Ensembl conventions.

I agree that the best solution is to get an Ensembl annotation that has UCSC headers rather than re-aligning the reads, that would be unnecessary. I'm planning to post all of our UCSC-derived Ensembl annotations on the MISO website soon. All I did was fetch the UCSC tables and then convert them to GTF (as described on website) and then to GFF3. Our experimental RNA-Seq pipeline rnaseqlib (see http://yarden.github.com/rnaseqlib and http://rnaseqlib.readthedocs.org/en/latest/#initializing-an-rna-base-for-your-genome) does this for you, by the way, for any UCSC genome.
